test/suite/test_search_near02.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+import time, wiredtiger, wttest, unittest
+from wtscenario import make_scenarios
+
+# test_search_near02.py
+# Test various prefix search near scenarios.
+class test_search_near02(wttest.WiredTigerTestCase):
+    key_format_values = [
+        ('fixed_string', dict(key_format='10s')),
+        ('var_string', dict(key_format='S')),
+        ('byte_array', dict(key_format='u')),
+    ]
+
+    scenarios = make_scenarios(key_format_values)
+
+    def test_search_near(self):
+        uri = 'table:test_base_scenario'
+        self.session.create(uri, 'key_format={},value_format=S'.format(self.key_format))
+        cursor = self.session.open_cursor(uri)
+        cursor2 = self.session.open_cursor(uri, None, "debug=(release_evict=true)")
+
+        # Basic character array.
+        l = "abcdefghijklmnopqrstuvwxyz"
+
+        # Insert keys aaa -> aaz with timestamp 200.
+        prefix = "aa"
+        self.session.begin_transaction()
+        for k in range (0, 25):
+            cursor[prefix + l[k]] = prefix + l[k]
+        cursor[prefix] = prefix
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(200))
+
+        # Insert key aaz timestamp 50.
+        self.session.begin_transaction()
+        cursor[prefix + "z"] = prefix + "z"
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(50))
+
+        # Evict the whole range.
+        for k in range (0, 26):
+            cursor2.set_key(prefix + l[k])
+            ret = cursor2.search()
+            cursor2.reset()
+
+        # Start a transaction at timestamp 100, aaz should be the only key that is visibile.
```

Comment:
Should we repeat that section for ts < 50, 50 < ts < 200 and ts > 200 ?

Reply:
For cases ts < 50 or ts > 200, is there scenarios in particular you have in mind that we should test?

Reply:
If ts < 50, we expect WT_NOTFOUND from search_near
If ts > 200, we expect search_near to return 1 and the cursor to be on the key aaa.

Reply:
thanks! added those in.